_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_ddi.cpp

```diff
@@ -45,12 +45,20 @@ GUID GetGUID(MfxVideoParam const & par)
 
     cfId = mfx::clamp<mfxU16>(par.m_ext.CO3.TargetChromaFormatPlus1 - 1, MFX_CHROMAFORMAT_YUV420, MFX_CHROMAFORMAT_YUV444) - MFX_CHROMAFORMAT_YUV420;
 
-    if (par.m_platform && par.m_platform < MFX_HW_ICL)
-        cfId = 0; // platforms below ICL do not support Main422/Main444 profile, using Main instead.
-    else if (par.m_platform && par.m_platform == MFX_HW_ICL_LP
-            && !cFamily
-            && cfId > 1)
-        cfId = 1; // ICL does not support Main444 profile without VDEnc
+    if (par.m_platform)
+    {
+        if (par.m_platform < MFX_HW_ICL)
+        {
+            cfId = 0; // platforms below ICL do not support Main422/Main444 profile, using Main instead.
+        }
+        else
+        {
+            if (cFamily == 0 && cfId > 1)
+                cfId = 1; // ICL+ platforms do not support Main444 profile without VDEnc
+            if (cFamily > 0 && cfId == 1)
+                cfId = 0; // ICL+ platforms do not support Main422 profile on VDEnc
+        }
```

Comment:
Can you please clarify why we need change color format?
I mean that if user ask for 422 encode with lowpower, why we don`t say  `MFX_ERR_UNSUPPORTED` in query, instead of change color format to 420?

Reply:
If we don't change it we can't create device and can't reach CheckVideoParam func

Reply:
I would agree with Anton. This part of code is to choose guid from input params.
We need this guid only to check if it is in the list of guids supported on the device.
If it is there we can create corresponding VideoDecoder.
Is it correct?
If yes then we will create incorrect VedeoDecoder with your change.

Reply:
I see two ways:
1. replace color format and further send it to query CheckVideoParam
2. don't change here but do like this commit https://github.com/Intel-Media-SDK/MediaSDK/commit/fd76d781ba3bc6f874a7c43b96ada291921c565f in CreateAccelerationService func to handle returned status

Reply:
@saosipov , please reply

Reply:
@saosipov , ping